astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -296,6 +296,18 @@ def test_power_array_array(self):
     def test_power_array_array2(self):
         np.power([2., 4.] * u.m, [2., 4.])
 
+    def test_power_array_array3(self):
+        # Equivalent unit fractions are NOT considered dimensionless and raise
+        # a ValueError.
+        with raises(ValueError):
+            np.power([2., 4.] * u.m / u.cm, [2., 4.])
+
+        # But identical unit fractions are considered dimensionless and are
+        # allowed as base for np.power: #4764
+        res = np.power([2., 4.] * u.m / u.m, [2., 4.])
```

Comment:
:+1: Yeah, I pulled your PR down and confirmed that those do work, as you claimed. I am neutral to whether to update this test or not. I'll let you, @mhvk , and @adonath hash it out.